### DISCUSSION TITLE
Missing GitHub Pull Requests and Issues extension

### DISCUSSION THREAD

**Discussion by maiertech** [community/community#26352] [Codespaces]
<p>I have a config in my <code>settings.json</code> called <code>codespaces.defaultExtensions</code> which includes <code>github.vscode-pull-request-github</code>. Normally, when I launch a Codespace, the extension is installed and ready to use.</p>
<p>Since today, however this does not work properly any more. When I launch my Codespace in Google Chrome, everything works fine. I see GitHub extensions icon on the left hand side, I can click on it and interact as expected. See this screenshot:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/8/b8f9bbde956a3295b058c17a7e112008ecbfb735.png" data-download-href="/uploads/short-url/qomUe7TDucdUF2ofgZerMWoOSK9.png?dl=1" title="Screen Shot 2020-11-21 at 1.44.50 AM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1482402/181097281-89001df7-1f12-4a4d-8d4d-33a5dd1f1023.png" alt="Screen Shot 2020-11-21 at 1.44.50 AM" data-base62-sha1="qomUe7TDucdUF2ofgZerMWoOSK9" width="408" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/8/b8f9bbde956a3295b058c17a7e112008ecbfb735_2_408x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/8/b8f9bbde956a3295b058c17a7e112008ecbfb735_2_612x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/b/8/b8f9bbde956a3295b058c17a7e112008ecbfb735_2_816x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/8/b8f9bbde956a3295b058c17a7e112008ecbfb735_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-11-21 at 1.44.50 AM</span><span class="informations">1030×1260 57.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>But when I launch the same Codespace inside VSCode, the extension’s icon goes missing on the left hand side, see this screenshot:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/6/861830d6daa52a3a502ce063e10e6c64b331bad0.png" data-download-href="/uploads/short-url/j8fObSYz3MNRIuOfCUR2tlnuORi.png?dl=1" title="Screen Shot 2020-11-21 at 1.47.26 AM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1482402/181097283-7f94176e-9f07-454d-abc8-443db91128a7.png" alt="Screen Shot 2020-11-21 at 1.47.26 AM" data-base62-sha1="j8fObSYz3MNRIuOfCUR2tlnuORi" width="226" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/8/6/861830d6daa52a3a502ce063e10e6c64b331bad0_2_226x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/8/6/861830d6daa52a3a502ce063e10e6c64b331bad0_2_339x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/8/6/861830d6daa52a3a502ce063e10e6c64b331bad0_2_452x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/6/861830d6daa52a3a502ce063e10e6c64b331bad0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-11-21 at 1.47.26 AM</span><span class="informations">576×1270 50.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>When looking for <code>github.vscode-pull-request-github</code> in the extensions view, the extension shows up grayed-out and is marked as “Built-in”. I can’t activate or deactivate the extension or install or uninstall it. See this screenshot:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/6/866018378bda1d7bce722e281308bcc047a23654.jpeg" data-download-href="/uploads/short-url/jaJRuthy36xVpOHgkIG8ZUSQGMc.jpeg?dl=1" title="Screen Shot 2020-11-21 at 1.49.08 AM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1482402/181097284-5ab1305c-1fad-4f77-a897-2a3a5a33f1f5.jpeg" alt="Screen Shot 2020-11-21 at 1.49.08 AM" data-base62-sha1="jaJRuthy36xVpOHgkIG8ZUSQGMc" width="690" height="272" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/8/6/866018378bda1d7bce722e281308bcc047a23654_2_690x272.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/8/6/866018378bda1d7bce722e281308bcc047a23654_2_1035x408.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/8/6/866018378bda1d7bce722e281308bcc047a23654_2_1380x544.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/6/866018378bda1d7bce722e281308bcc047a23654_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-11-21 at 1.49.08 AM</span><span class="informations">2510×992 338 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I also could not figure out a way to bring it’s icon back on the left.</p>
<p>What’s going wrong here?</p>

**Comment by jkeech** [comment]
<p>There’s a set of core functionality for Codespaces that we are starting to bundle with Codespaces, including the Codespaces extension and the Pull Requests extension. In order to keep the default extension view less cluttered and prevent users from accidentally uninstalling those core extensions, which would break some Codespaces functionality, we are switching to mark those extensions as built-in. They should show up if you include <code>@builtin</code> in the extensions viewlet query, but it’s expected that it cannot be uninstalled.</p>
<p>It’s not expected that the extension fails to activate. Do you see anything in the <code>GitHub Pull Request</code> output window? Also, which VS Code version are you using? I just tried on a few different Codespaces, and the extension is showing up for me when connected from VS Code on the desktop.</p>
<p>By the way, are you using the new <a href="https://marketplace.visualstudio.com/items?itemName=GitHub.codespaces">GitHub Codespaces</a> extension or Visual Studio Codespaces? GitHub Codespaces was published this week, and we’ll be switching to web to install that one by default over the next couple weeks, and GitHub Codespaces is where new features are being added. If you don’t need to use Visual Studio Codespaces at all, I would recommend uninstalling that extension and installing GitHub Codespaces. Both extensions can run side-by-side, but you’ll see some duplicated commands in the command palette, so it’s the best experience to just use GitHub Codespaces going forward.</p>

**Comment by maiertech** [comment]
<p>I switched over to the GitHub Codespaces extension, as you suggested, and uninstalled the Visual Studio Codespaces extensions.</p>
<p>As of the GitHub Pull Requests extension, it is somehow stuck with a “Reload Required” button, and pressing the button reloads VS Code, but does not make the reload button go away:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/0/402beff96a2d1f89a0ded8857560a8875b94670b.png" data-download-href="/uploads/short-url/99GH016ZYUBG2IDMxud41k1HJIT.png?dl=1" title="Screen Shot 2020-11-21 at 10.54.41 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1482402/181097286-8f701e36-72a1-4e12-915d-1ae4425cb31d.png" alt="Screen Shot 2020-11-21 at 10.54.41 PM" data-base62-sha1="99GH016ZYUBG2IDMxud41k1HJIT" width="243" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/0/402beff96a2d1f89a0ded8857560a8875b94670b_2_243x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/0/402beff96a2d1f89a0ded8857560a8875b94670b_2_364x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/4/0/402beff96a2d1f89a0ded8857560a8875b94670b_2_486x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/0/402beff96a2d1f89a0ded8857560a8875b94670b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-11-21 at 10.54.41 PM</span><span class="informations">684×1402 116 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>This happens in VS Code 1.51.1. The extension is at 0.21.2.</p>
<p>I just double checked and the issue does not happen in VS Code 1.52.0 (Insiders).</p>

**Comment by maiertech** [comment]
<p>And it’s broken again in this one:</p>
<p>Version: 1.52.0-insider<br>
Commit: ac165d7f974d70212be80cb15eaecfc1fd20b21a<br>
Date: 2020-11-30T17:21:33.245Z<br>
Electron: 11.0.2<br>
Chrome: 87.0.4280.63<br>
Node.js: 12.18.3<br>
V8: 8.7.220.24-electron.0<br>
OS: Darwin x64 20.1.0</p>

**Comment by maiertech** [comment]
<p>Unfortunately, the latest insider</p>
<p>Version: 1.52.0-insider<br>
Commit: bf21395291146c014220eb70b54fdfc0a15956b5<br>
Date: 2020-12-01T09:27:28.568Z<br>
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 11_0_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.67 Safari/537.36</p>
<p>in the browser version of Codespaces breaks the <code>github.vscode-pull-request-github</code> extension, see screenshot:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/c/6cd1ceb8455e4246faba92a2306db496051f7654.png" data-download-href="/uploads/short-url/fwF4N9m4PJz0fqqvYqKMB4S67lO.png?dl=1" title="Screen Shot 2020-12-01 at 11.46.33 AM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1482402/181097291-0e7bac3a-a185-4ce3-82e9-fc9bb6856408.png" alt="Screen Shot 2020-12-01 at 11.46.33 AM" data-base62-sha1="fwF4N9m4PJz0fqqvYqKMB4S67lO" width="303" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/c/6cd1ceb8455e4246faba92a2306db496051f7654_2_303x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/6/c/6cd1ceb8455e4246faba92a2306db496051f7654_2_454x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/6/c/6cd1ceb8455e4246faba92a2306db496051f7654_2_606x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/c/6cd1ceb8455e4246faba92a2306db496051f7654_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-12-01 at 11.46.33 AM</span><span class="informations">774×1276 29.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>This is an insider build issue and I opened this issue: <a href="https://github.com/microsoft/vscode-pull-request-github/issues/2304" rel="noopener nofollow ugc">https://github.com/microsoft/vscode-pull-request-github/issues/2304</a>. But the question is if you can roll back the browser VSCode to the last insider which does not break this (I think 29 Nov 2020)?</p>

**Comment by maiertech** [comment]
<p>This is fixed again in the insider build with <code>github.vscode-pull-request-github</code> 0.21.4. You need to discard your Codespace and create a new one.</p>

**Comment by jkeech** [comment, marked as answer]
<p><a class="mention" href="/454de6e">@454de6e</a>, thanks for reporting the issue. Sorry you have been having so many problems recently. I can confirm that the newly released version 0.21.4 should fix the problem. I don’t think you should have to recreate the Codespace to update, but recreating would also work. Let us know if you continue to run into issues on 0.21.4. Thanks!</p>
